### ISSUE TITLE
VersionedTransaction.from_json Not Working

### ISSUE THREAD

**Issue by Dassy23** [kevinheavey/solders#67]
I get the following error when trying to convert a tx to json and back 
 
 ValueError: invalid type: integer `128`, expected message prefix byte at line 1 column 164

```
url = f'https://quote-api.jup.ag/v6/quote?inputMint=So11111111111111111111111111111111111111112&outputMint=EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v&amount=1000000&slippageBps=1'
resp = requests.get(url)
quote = resp.json()
swap_transaction = requests.post(
    'https://quote-api.jup.ag/v6/swap',
    headers={'Content-Type': 'application/json'},
    json={
        'quoteResponse': quote,
        'userPublicKey': authority,
        'wrapAndUnwrapSol': True
    }
).json()
swapTransactionBuf = base64.b64decode(swap_transaction['swapTransaction'])
tx = VersionedTransaction.from_bytes(swapTransactionBuf)
t = tx.to_json()
t1 = VersionedTransaction.from_json(t)
   
```

**Comment by Dassy23** [comment]
packing a new versiontx with a messagev0 is a good way im getting around this now

```
 msg = MessageV0.from_json(json.dumps(txn['message'][1]))
 signed_transaction = VersionedTransaction(msg,[payer.entity])
```

**Comment by kevinheavey** [comment]
can you post the raw transaction bytes please? This thing: `swapTransactionBuf`

**Comment by Dassy23** [comment]
yupp see below 

```
b"\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x01\x00\x08\r\x05\xd6]S\x86q\xf1I6\x04!\xd3\xbe\x18u\x13\xb7\x14uCY\x8fU\xfcl`msq.M\x86\x0bH\x07Ou\xb8\xea0i\xc9\xed?\x06\xbc2\x19x*\xe4\xcd\x8e\\\x9d7I\x82V\xb1\x98K\x9dPB\xe4\xb9SIZ4\x84d\xb3n\tf\x08\x87\x96\xf8TBW}@\x12\x85\x15\xd5\x0b\xd9[\x85\t\xc2X\xefg\x7f\xb5c^dsrKp\xe1kd\x05T\x03N\xa4z\x1c{?\xcd\x88\x85<A]2T\xfeF\xcb\xe6\x01\xbc\xa5\n\xf2\xfaIR\xb8U\xf0\x82\xe4\xe9\xb2>m\xd4\x00\xbe\x91 ;\x81\x1c\xc2\x05\xd9\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x06Fo\xe5!\x172\xff\xec\xad\xbar\xc3\x9b\xe7\xbc\x8c\xe5\xbb\xc5\xf7\x12k,C\x9b:@\x00\x00\x00\x04y\xd5[\xf21\xc0n\xeet\xc5n\xceh\x15\x07\xfd\xb1\xb2\xde\xa3\xf4\x8eQ\x02\xb1\xcd\xa2V\xbc\x13\x8f\x06\xdd\xf6\xe1\xd7e\xa1\x93\xd9\xcb\xe1F\xce\xeby\xac\x1c\xb4\x85\xed_[7\x91:\x8c\xf5\x85~\xff\x00\xa9:\xb8\x90?\xb75\xca\xb1\xc6|Y\xafHW\xed\xf6\x1b\n\xf82\xa5\n|Y\xe3!\x91\x9e\x0e\xc8\xa9\xbc\x8c\x97%\x8fN$\x89\xf1\xbb=\x10)\x14\x8e\r\x83\x0bZ\x13\x99\xda\xff\x10\x84\x04\x8e{\xd8\xdb\xe9\xf8Y\xb4?\xfa'\xf5\xd7\xf6Jt\xc0\x9b\x1f)Xy\xdeK\t\xab6\xdf\xc9\xddQK2\x1a\xa7\xb3\x8c\xe5\xe8\xc6\xfaz\xf3\xbe\xdb\xad:=e\xf3j\xab\xc9t1\xb1\xbb\xe4\xc2\xd2\xf6\xe0\xe4|\xa6\x02\x03E/]a9\x10\xb1a&\xa2yB,tTx\x02\x82\xbd\x86t\x9a\x8c\n|9\xeaJ\xc4\xfe\xc3\xd3fv\x94\x0f\x07\x06\x00\x05\x02\xc0\\\x15\x00\x06\x00\t\x03\x10'\x00\x00\x00\x00\x00\x00\n\x06\x00\x02\x00\x1c\x05\x08\x01\x01\x05\x02\x00\x02\x0c\x02\x00\x00\x00\x00\xe4\x0bT\x02\x00\x00\x00\x08\x01\x02\x01\x11\x07$\x08\t\x00\x02\x03\x01\x04\x1c\x0c\x07\x07\x0b\x07\x16\x1b\x10\t\x03\x01\x0f\x12\x14\x13\x08\x19\x1a\x18\x15\x17\x0e\t\x03\x01\x11\r\x08)\xc1 \x9b3A\xd6\x9c\x81\x04\x02\x00\x00\x00\x19(\x00\x02\x1c\x01<\x00\x02\x00\xe4\x0bT\x02\x00\x00\x00\x0b\xeb\xb30\x00\x00\x00\x00\x01\x00\x00\x08\x03\x02\x00\x00\x01\t\x02\xfa\x9f\xf6v\xf1\xeb\nLh\x9f:\xa3\xff|AxJ\\\xea\xc5J]:\x03!?KE^\xa6\xa68\x08\x02\x03@A\x05<?=\x07\x04\x13\x06>B\x0b;\xcc'\xedks\xaf\x15O\xdb}@\x0f\xb2\xff\xccQ\xd4\x7f\x15}}\xdepa\xcft\x96\xae\xfd<dj\x00\x01]"
```

**Comment by kevinheavey** [comment]
The issue lies in the Solana Rust SDK: https://github.com/solana-labs/solana/issues/34608

**Comment by kevinheavey** [comment]
Fixed by #80
